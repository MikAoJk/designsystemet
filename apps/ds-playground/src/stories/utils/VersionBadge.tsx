import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version = process.env[`@skatteetaten/ds-${packageName}`] ?? '..';
  if (process.env['NODE_ENV'] === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
