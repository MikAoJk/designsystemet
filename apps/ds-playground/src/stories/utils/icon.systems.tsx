import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
