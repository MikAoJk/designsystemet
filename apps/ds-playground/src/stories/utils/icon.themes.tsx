import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
