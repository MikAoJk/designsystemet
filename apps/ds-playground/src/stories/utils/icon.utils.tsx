import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
