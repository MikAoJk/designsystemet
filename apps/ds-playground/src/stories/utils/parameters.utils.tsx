import { ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
import type { Parameters } from '@storybook/csf';

type DesignParameters = {
  name: string;
  url: string;
};

export const getParameters = function (
  designParameters: DesignParameters[]
): Parameters {
  return {
    docs: {
      page: (): JSX.Element => {
        return (
          <>
            <Stories title={''} />
            <ArgsTable story={PRIMARY_STORY} />
          </>
        );
      },
    },
    design: designParameters.map((designParameter: DesignParameters) => {
      return {
        name: designParameter.name,
        type: 'figma',
        url: designParameter.url,
      };
    }),
    backgrounds: {
      values: [
        { name: 'Svart', value: '#000' },
        { name: 'Hvit', value: '#fff' },
      ],
    },
  };
};
