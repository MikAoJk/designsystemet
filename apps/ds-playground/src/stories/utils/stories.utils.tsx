export const exampleParameters = {
  docs: {
    source: {
      type: 'code',
      language: 'tsx',
      transform: (source: string): string => {
        return source
          .replace('render: (_args): JSX.Element', 'const Examples = ()')
          .replace(
            'render: function Render(_args): JSX.Element ',
            'const Examples = () => '
          )
          .replace('satisfies Story', '');
      },
    },
  },
  /* _args i en story må finnes slik at props i fanen Controls skjules og er prefikset med underscore for å unngå eslint klaging */
  controls: {
    exclude: /.*/,
  },
};
