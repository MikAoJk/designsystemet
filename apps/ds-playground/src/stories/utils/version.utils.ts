export const getVersion = (
  packageName: string
): {
  major: string;
  minor: string;
  patch: string;
  postfix?: string;
  style?: Record<string, string>;
} => {
  const ver = process.env[`@skatteetaten/${packageName}`] ?? '..';
  const [major, minor, patch] = ver.split('.');
  const majorWithPackageName = packageName ? `${packageName}@${major}` : major;

  const style = {
    color: 'var(--semantic-page-foreground)',
    background: 'var(--semantic-success-background)',
    'font-weight': 'var(--font-weight-bold)',
    'font-size': '13px',
  };

  if (process.env['NODE_ENV'] === 'development') {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: 'develop',
      style: {
        ...style,
        background: 'var(--semantic-danger-background)',
      },
    };
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: process.env['STORYBOOK_BUILD_VERSION'],
      style: {
        ...style,
        background: 'var(--semantic-warning-background)',
      },
    };
  }

  return {
    major: majorWithPackageName,
    minor,
    patch,

    style: {
      ...style,
    },
  };
};
