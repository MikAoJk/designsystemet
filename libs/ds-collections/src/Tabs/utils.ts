export const valueRegex = /^[\w:.-_]*$/;
