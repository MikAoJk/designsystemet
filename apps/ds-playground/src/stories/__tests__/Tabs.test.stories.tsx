import { Tabs, TabsProps } from '@skatteetaten/ds-collections';
import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { expect, fn, userEvent, waitFor, within } from '@storybook/test';

import { wrapper } from './testUtils/storybook.testing.utils';

const meta = {
  component: Tabs,
  title: 'Tester/Tabs/Tabs',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    defaultValue: { table: { disable: true } },
    value: { table: { disable: true } },
    isMultiline: { table: { disable: true } },
    variant: { table: { disable: true } },
    hasBorder: { table: { disable: true } },
    // Events
    onChange: { table: { disable: true } },
  },
} satisfies Meta<typeof Tabs>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: TabsProps = {
  defaultValue: 'tab1',
};

const TemplateTabs: StoryFn<typeof Tabs> = (args) => {
  return (
    <Tabs {...args}>
      <Tabs.List>
        <Tabs.Tab value={'tab1'}>{'Person'}</Tabs.Tab>
        <Tabs.Tab value={'tab2'}>{'Bedrift'}</Tabs.Tab>
        <Tabs.Tab value={'tab3'}>{'Rettskilder'}</Tabs.Tab>
      </Tabs.List>
      <Tabs.Panel value={'tab1'}>{'Tabs.Panel Person'}</Tabs.Panel>
      <Tabs.Panel value={'tab2'}>{'Tabs.Panel Bedrift'}</Tabs.Panel>
      <Tabs.Panel value={'tab3'}>{'Tabs.Panel Rettskilder'}</Tabs.Panel>
    </Tabs>
  );
};

export const Defaults = {
  name: 'Defaults (A2)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    defaultValue: 'tab1',
  },
  argTypes: {
    defaultValue: { table: { disable: false } },
  },
  parameters: {},
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const tab = canvas.getByRole('tab', { name: 'Person' });
    await expect(tab).toBeInTheDocument();
  },
} satisfies Story;

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: { imageSnapshot: { disable: true } },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const tabs = canvas.getAllByRole('generic')[1];
    await expect(tabs).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithVariantCompact = {
  name: 'With Variant Compact (A1)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    defaultValue: 'tab1',
    variant: 'compact',
  },
  parameters: {
    imageSnapshot: { disable: false },
  },
  /*   play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const firstTab = canvas.getByRole('tab', {
      name: 'Person',
    });
    await expect(firstTab).toBeInTheDocument();
    await expect(firstTab).toHaveAttribute('aria-selected', 'true');
  }, */
} satisfies Story;

export const WithTabSelected = {
  name: 'With Tab Click (A2)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    defaultValue: 'tab1',
    onChange: fn(),
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ args, canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const firstTab = canvas.getByRole('tab', {
      name: 'Person',
    });
    await expect(firstTab).toBeInTheDocument();
    await expect(firstTab).toHaveAttribute('aria-selected', 'true');
    const secondTab = canvas.getByRole('tab', {
      name: 'Bedrift',
    });
    await userEvent.click(secondTab);
    await waitFor(() => expect(args.onChange).toHaveBeenCalled());
  },
} satisfies Story;

export const WithIcon = {
  name: 'With Icon (A4)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    defaultValue: 'tab1',
  },
  parameters: {
    imageSnapshot: {
      focus: `${wrapper} > tab`,
      hover: `${wrapper} > tab`,
      click: `${wrapper} > tab`,
    },
  },
  /*   play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const firstTab = canvas.getByRole('tab', {
      name: 'Person',
    });
    await expect(firstTab).toBeInTheDocument();
    await expect(firstTab).toHaveAttribute('aria-selected', 'true');
  }, */
} satisfies Story;
