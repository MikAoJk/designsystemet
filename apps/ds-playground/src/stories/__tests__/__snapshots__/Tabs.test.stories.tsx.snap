// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Tabs/Tabs Defaults play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r0:-tab1"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r0:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r0:-tab2"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r0:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r0:-tab3"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r0:-tab3"
      >
        Rettskilder
      </button>
    </div>
    <div id="ds-tab-panel-:r0:-tab1"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r0:-tab2"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r0:-tab3"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab3"
    >
      Tabs.Panel Rettskilder
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithAriaRolesTabindex play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r6:-tab1"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r6:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r6:-tab2"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r6:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r6:-tab3"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r6:-tab3"
      >
        Rettskilder
      </button>
    </div>
    <div id="ds-tab-panel-:r6:-tab1"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r6:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r6:-tab2"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r6:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r6:-tab3"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r6:-tab3"
    >
      Tabs.Panel Rettskilder
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       class="dummyClassname"
       lang="nb"
       data-testid="123ID"
  >
    <div class="TabsList_tabList__vWYZt  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r2:-tab1"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r2:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r2:-tab2"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r2:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r2:-tab3"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r2:-tab3"
      >
        Rettskilder
      </button>
    </div>
    <div id="ds-tab-panel-:r2:-tab1"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r2:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r2:-tab2"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r2:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r2:-tab3"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r2:-tab3"
    >
      Tabs.Panel Rettskilder
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithDefaultValue play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r4:-tab1"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r4:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r4:-tab2"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r4:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r4:-tab3"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r4:-tab3"
      >
        Rettskilder
      </button>
    </div>
    <div id="ds-tab-panel-:r4:-tab1"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r4:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r4:-tab2"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r4:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r4:-tab3"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r4:-tab3"
    >
      Tabs.Panel Rettskilder
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithMultiline smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt TabsList_tabList_multiline__EPlby "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r7:-tab1"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r7:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r7:-tab2"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r7:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r7:-tab3"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r7:-tab3"
      >
        Ubehandlede saker (+99)
      </button>
      <button id="ds-tab-id-:r7:-tab4"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r7:-tab4"
      >
        Behandlede saker (1)
      </button>
    </div>
    <div id="ds-tab-panel-:r7:-tab1"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab1"
    >
      Tabs.Panel - Person
    </div>
    <div id="ds-tab-panel-:r7:-tab2"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab2"
    >
      Tabs.Panel - Bedrift
    </div>
    <div id="ds-tab-panel-:r7:-tab3"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab3"
    >
      Tabs.Panel - Ubehandlede saker (+99)
    </div>
    <div id="ds-tab-panel-:r7:-tab4"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab4"
    >
      Tabs.Panel - Behandlede
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithRef play-test 1`] = `
<div data-test-block="true">
  <div class
       id="dummyIdForwardedFromRef"
  >
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithTabClick play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r8:-tab1"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r8:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r8:-tab2"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r8:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r8:-tab3"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r8:-tab3"
      >
        Rettskilder
      </button>
    </div>
    <div id="ds-tab-panel-:r8:-tab1"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r8:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r8:-tab2"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r8:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r8:-tab3"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r8:-tab3"
    >
      Tabs.Panel Rettskilder
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithValue play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r5:-tab1"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r5:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r5:-tab2"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r5:-tab2"
      >
        Bedrift
      </button>
    </div>
    <div id="ds-tab-panel-:r5:-tab1"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r5:-tab1"
    >
      Tabs.Panel - Person
    </div>
    <div id="ds-tab-panel-:r5:-tab2"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r5:-tab2"
    >
      Tabs.Panel - Bedrift
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithVariantCompact smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r3:-tab1"
              class="TabsTab_tab__W3ulY TabsTab_tab_compact__iTTkr  TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r3:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r3:-tab2"
              class="TabsTab_tab__W3ulY TabsTab_tab_compact__iTTkr    "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r3:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r3:-tab3"
              class="TabsTab_tab__W3ulY TabsTab_tab_compact__iTTkr    "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r3:-tab3"
      >
        Rettskilder
      </button>
    </div>
    <div id="ds-tab-panel-:r3:-tab1"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r3:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r3:-tab2"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r3:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r3:-tab3"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r3:-tab3"
    >
      Tabs.Panel Rettskilder
    </div>
  </div>
</div>
`;
